homeassistant/components/rfxtrx/__init__.py

```diff
@@ -224,6 +228,25 @@ def _add_device(event, device_id):
         hass.config_entries.async_update_entry(entry=entry, data=data)
         devices[device_id] = config
 
+    @callback
+    def _remove_device(event: Event):
+        if event.data["action"] != "remove":
+            return
+        device_entry = device_registry.deleted_devices[event.data["device_id"]]
```

Comment:
Why not do the work in `async_remove_config_entry_device`? We already have the device entry there.

Reply:
Thought the device might have some other way to get removed, so it would be good to act on the event instead. This is how the only other user of this does it (cast). I can move it.

Reply:
There was one more reason. It need access to the `devices` variable, which is currently local to to that function. So can't easily move it to the other function without larger refactoring.

Reply:
Ok.